controller/server/update.ml

```diff
@@ -149,16 +154,23 @@ let rec run ~update_url ~rauc ~set_state =
         let inactive_up_to_date = inactive_version_compare == 0 in
         let inactive_update_available = inactive_version_compare > 0 in
 
-        if booted_up_to_date then
-          (* Don't care if inactive can be updated. I.e. Only update the inactive partition once the booted partition is outdated. This results in always two versions being available on the machine. *)
-          UpToDate version_info
-          |> set
-
-        else if inactive_up_to_date then
-          (* If booted is not up to date but inactive is up to date, we probably should reboot *)
-          (* TODO this is where you should check if inactive boot partition has more than 3 boot attempts, if it does, you probably should reinstall. *)
-          RebootRequired
-          |> set
+        if booted_up_to_date || inactive_up_to_date then
+          match%lwt Rauc.get_primary rauc with
+          | Some primary_slot ->
+            if booted_up_to_date then
+              (* Don't care if inactive can be updated. I.e. Only update the inactive partition once the booted partition is outdated. This results in always two versions being available on the machine. *)
+              UpToDate version_info |> set
+            else
+              let%lwt booted_slot = Rauc.get_booted_slot rauc in
+              if booted_slot == primary_slot then
+                (* Inactive is up to date while booted is out of date, but booted was specifically selected for boot *)
+                OutOfDateVersionSelected |> set
+              else
+                (* If booted is not up to date but inactive is up to date, we probably should reboot *)
+                RebootRequired |> set
```

Comment:
The comment could be more precise:

```ocaml
(* Inactive is up to date while booted is out of date, and booted was not specifically selected for boot *)
```

Though, is it possible that booted was not specifically selected for boot, i.e. `booted_slot != primary_slot` ?

Reply:
You're right, I left the comment that was there before, but it's not 100 % adequate now. I will update it.

`booted_slot != primary_slot` at this moment can only happen after an update has just been applied. Then e.g. system A is still booted, but system B has been made primary.